app/javascript/components/tree-view/hierarchical-tree-view.jsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Tree, { ActionTypes } from '@manageiq/react-ui-components/dist/wooden-tree';
+
+class HierarchicalTreeView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tree: HierarchicalTreeView.convertData(props.bs_tree)
+    }
+  };
+
+  static convertData = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(JSON.parse(data)));
+
+  actionMapper = () => ({
```

Comment:
I don't see purpose of this being a function. There are no arguments which would change it. You can replace this just by an object.